### PR TITLE
Refactoring lockin binaries

### PR DESCRIPTION
**Pull request by ryan-summers**
This PR merges `lockin-external` and `lockin-internal` into a single binary.

This has also refactored the output configuration to make it more configurable. This was needed to work well with the reference signal generation, as that output should necessarily just override a different setting.

### CODE REVIEW

File: src/bin/lockin.rs [hunk 1]
```diff
@@ -18,16 +18,34 @@ use stabilizer::hardware::{
...
+const DAC_SEQUENCE: [i16; design_parameters::SAMPLE_BUFFER_SIZE] =
+    [ONE, SQRT2, 0, -SQRT2, -ONE, -SQRT2, 0, SQRT2];
+
 #[derive(Copy, Clone, Debug, Deserialize, Miniconf)]
 enum Conf {
-    PowerPhase,
+    Power,
+    Phase,
+    PllFrequency,
     FrequencyDiscriminator,
```
Comment (ryan-summers) [inline src/bin/lockin.rs:33]: @jordens Do you have better names than `PllFrequency` and `FrequencyDescriminator`? I don't know if the names I selected make sense for these measurements (as I'm not quite sure what these values are).

Comment (jordens) [reply src/bin/lockin.rs:33]: `InPhase` (`.re`), `Quadrature` (`.im`), `Magnitude` (`abs_sqr()`), `LogPower` (`log2()`), `Phase` (`arg()`), `PllFrequency` (`pll_frequency`), `Modulation` (`DAC_SEQUENCE`) should be the ones. Each channel should support all of them, independently.

Comment (ryan-summers) [reply src/bin/lockin.rs]: Alright, should now be cleaned up. Thanks for the input!

**Comment by ryan-summers** [comment]
Alright, testing on Stabilizer with an oscilloscope shows that both channels are independently configurable and `miniconf.py` shows them swapping over measurements as requested.

@jordens This is ready for review whenever you get a chance.

File: src/bin/lockin.rs [hunk 2]
```diff
@@ -137,29 +157,57 @@ const APP: () = {
...
-            .lockin_phase
-            .wrapping_add(pll_phase.wrapping_mul(settings.lockin_harmonic));
+                // Demodulation LO phase offset
+                let phase_offset: i32 = 1 << 30;
+
+                let sample_frequency = (pll_frequency as i32)
+                    .wrapping_mul(settings.lockin_harmonic);
+                let sample_phase = phase_offset.wrapping_add(
+                    pll_phase.wrapping_mul(settings.lockin_harmonic),
+                );
```
Comment (jordens) [inline src/bin/lockin.rs:206]: This should be unified with L182-L188, use `settings.phase_offset` as well, and pulled out of the `match`. But I can do that later.

File: src/bin/lockin.rs [hunk 3]
```diff
@@ -137,29 +157,57 @@ const APP: () = {
...
-            settings.pll_tc[0],
-            settings.pll_tc[1],
-        );
+        let mut pll_frequency = 0;
+
+        let (sample_phase, sample_frequency) = match settings.lockin_mode {
+            LockinMode::External => {
+                let timestamp =
+                    c.resources.timestamper.latest_timestamp().unwrap_or(None); // Ignore data from timer capture overflows.
+                let (pll_phase, frequency) = c.resources.pll.update(
```
Comment (jordens) [inline src/bin/lockin.rs:174]: Probably better to call these `reference_phase` and `reference_frequency` and let both arms return `reference_frequency` (constant in the `Internal` case) that's then used as `conf::ReferenceFrequency` output.

File: src/bin/lockin.rs [hunk 4]
```diff
@@ -18,16 +18,34 @@ use stabilizer::hardware::{
 use miniconf::Miniconf;
 use stabilizer::net::{Action, MqttInterface};
 
+// A constant sinusoid to send on the DAC output.
+// Full-scale gives a +/- 10.24V amplitude waveform. Scale it down to give +/- 1V.
+const ONE: i16 = ((1.0 / 10.24) * i16::MAX as f32) as _;
+const SQRT2: i16 = (ONE as f32 * 0.707) as _;
+const DAC_SEQUENCE: [i16; design_parameters::SAMPLE_BUFFER_SIZE] =
+    [ONE, SQRT2, 0, -SQRT2, -ONE, -SQRT2, 0, SQRT2];
```
Comment (jordens) [inline src/bin/lockin.rs:26]: Once #86 lands, this should be refactored to use that same signal generator.

**Review by jordens — APPROVED** [review]

**Comment by jordens** [comment]
bors merge

**Comment by ryan-summers** [comment]
bors cancel

**Comment by ryan-summers** [comment]
bors merge

**Comment by ryan-summers** [comment]
bors cancel

**Comment by ryan-summers** [comment]
bors merge

**Pull request merged**